src/sentry/seer/explorer/tools.py

```diff
@@ -0,0 +1,135 @@
+import logging
+from typing import Any
+
+from sentry.api import client
+from sentry.models.apikey import ApiKey
+from sentry.models.organization import Organization
+from sentry.snuba.referrer import Referrer
+
+logger = logging.getLogger(__name__)
+
+
+def execute_trace_query_chart(
+    *,
+    org_id: int,
+    query: str,
+    stats_period: str,
+    y_axes: list[str],
+    group_by: list[str] | None = None,
+) -> dict[str, Any] | None:
+    """
+    Execute a trace query to get chart/timeseries data by calling the events-stats endpoint.
+    """
+    try:
+        organization = Organization.objects.get(id=org_id)
+    except Organization.DoesNotExist:
+        logger.warning("Organization not found", extra={"org_id": org_id})
+        return None
+
+    # Get all project IDs for the organization
+    project_ids = list(organization.project_set.values_list("id", flat=True))
+    if not project_ids:
+        logger.warning("No projects found for organization", extra={"org_id": org_id})
+        return None
+
+    params: dict[str, Any] = {
```

Comment:
what do you think about moving more of this logic to the caller? so that way this can basically be a passthrough